src/modules/package.rs

```diff
@@ -223,44 +221,35 @@ fn get_sbt_version(context: &Context, config: &PackageConfig) -> Option<String>
 }
 
 fn get_cargo_version(context: &Context, config: &PackageConfig) -> Option<String> {
-    let mut file_contents = context.read_file_from_pwd("Cargo.toml")?;
+    let mut cargo_toml: toml::Table = context.read_file_from_pwd("Cargo.toml")?.parse().ok()?;
 
-    let mut cargo_toml: toml::Table = toml::from_str(&file_contents).ok()?;
     let cargo_version = cargo_toml.get("package").and_then(|p| p.get("version"));
+
     let raw_version = if let Some(v) = cargo_version.and_then(toml::Value::as_str) {
-        // regular version string
         v
-    } else if cargo_version
-        .and_then(|v| v.get("workspace"))
-        .and_then(toml::Value::as_bool)
-        .unwrap_or_default()
-    {
-        // workspace version string (`package.version.worspace = true`)
-        // need to read the Cargo.toml file from the workspace root
-        let mut version = None;
-        // discover the workspace root
-        for path in context.current_dir.ancestors().skip(1) {
-            // Assume the workspace root is the first ancestor that contains a Cargo.toml file
-            if let Ok(mut file) = fs::File::open(path.join("Cargo.toml")) {
-                file.read_to_string(&mut file_contents).ok()?;
-                cargo_toml = toml::from_str(&file_contents).ok()?;
-                // Read workspace.package.version
-                version = cargo_toml
-                    .get("workspace")?
-                    .get("package")?
-                    .get("version")?
-                    .as_str();
-                break;
+    } else if cargo_version.and_then(|v| v.get("workspace"))?.as_bool()? {
+        let mut ancestors = context
+            .current_dir
+            .ancestors()
+            .map(|a| a.join("Cargo.toml"))
+            .filter(|c| c.exists());
+
+        loop {
+            match cargo_toml
+                .get("workspace")
+                .and_then(|w| w.get("package"))
+                .and_then(|p| p.get("version"))
+                .and_then(|v| v.as_str())
+            {
+                Some(v) => break v,
+                None => {
+                    let cargo_path = ancestors.next()?;
+                    cargo_toml = std::fs::read_to_string(cargo_path).ok()?.parse().ok()?;
+                }
```

Comment:
I prefer the way this was handled previously, please just handle the initial case manually instead.